src/main/java/net/sf/jabref/logic/msbib/MSBibEntry.java

```diff
@@ -65,10 +65,11 @@
     private String bibtexEntryType;
 
     // reduced subset, supports only "CITY , STATE, COUNTRY"
-    // \b(\w+)\s?[,]?\s?(\w+)\s?[,]?\s?(\w+)\b
-    // WORD SPACE , SPACE WORD SPACE , SPACE WORD
-    // tested using http://www.javaregex.com/test.html
-    private static final Pattern ADDRESS_PATTERN = Pattern.compile("\\b(\\w+)\\s?[,]?\\s?(\\w+)\\s?[,]?\\s?(\\w+)\\b");
+    // \b(\w+)\s?[,]?\s?(\w+)\s?[,]?\s?(\w*)\b
+    // WORD SPACE , SPACE WORD SPACE (zero or more) , SPACE WORD (zero or word)
+    // tested using http://www.regexpal.com/
+    //Matches both single locations like Berlin or Stroudsburg, PA, USA
+    private static final Pattern ADDRESS_PATTERN = Pattern.compile("\\b(\\w+)\\s?[,]?\\s?(\\w*)\\s?[,]?\\s?(\\w*)\\b");
```

Comment:
Maybe we can think of documenting our regexes more. See DOI.java for an example.